script/20240516-revoke-roles/02_GrantAdminRoleForMultisig_Mainnet.s.sol

```diff
@@ -0,0 +1,362 @@
+    contracts[2] = address(_nameChecker);
+    contracts[3] = address(_rns);
+    contracts[4] = address(_auction);
+
+    vm.startBroadcast(duke);
+    // Transfer .ron domain ownership to owned multicaller
+    uint256[] memory ids = new uint256[](3);
+    ids[0] = 0x0;
+    ids[1] = LibRNSDomain.RON_ID;
+    ids[2] = LibRNSDomain.ADDR_REVERSE_ID;
+    _rns.setApprovalForAll(_batchTransfer, true);
+
+    // Bulk transfer .ron domain ownership to owned multicaller
+    (bool success, bytes memory returnOrRevertData) = _batchTransfer.call(
+      abi.encodeWithSignature("safeBatchTransfer(address,uint256[],address)", _rns, ids, _ownedMulticaller)
+    );
+    success.handleRevert(returnOrRevertData);
+
+    // Remove approval for batch transfer
+    _rns.setApprovalForAll(_batchTransfer, false);
+    _rns.setApprovalForAll(address(_auction), false);
+    _rns.setApprovalForAll(address(_ronController), false);
+    _rns.setApprovalForAll(address(_reverseRegistrar), false);
+    // Remove approval for Legacy Owned Multicaller
+    // Verify: https://app.roninchain.com/address/0x8975923D01132bEB6c412F827f63D44712726E13
+    _rns.setApprovalForAll(0x8975923D01132bEB6c412F827f63D44712726E13, false);
+    // Remove approval for Legacy RNS Operation contracts
+    _rns.setApprovalForAll(0xCD245263eDdEE593a5A66f93f74C58c544957339, false);
+    _rns.setApprovalForAll(0xd9b3CC879113C7ABaa7694d25801bFFD8Fae0F27, false);
+
+    uint256 length = contracts.length;
+
+    for (uint256 i; i < length; i++) {
+      AccessControlEnumerable(contracts[i]).grantRole(0x0, multisig);
+      console.log("Duke will renounce his admin roles of contract:", vm.getLabel(contracts[i]), "manually");
+
+      assertTrue(
+        AccessControlEnumerable(contracts[i]).getRoleMemberCount(0x0) > 1,
+        string.concat("Role is empty", "contract: ", vm.toString(contracts[i]))
+      );
+    }
+
+    console.log("Revoke roles for domain price", 0xAdc6a8fEB5C53303323A1D0280c0a0d5F2e1a14D);
+    // Remove another admin roles: https://sky-mavis.slack.com/archives/C06C3HW1HS7/p1712812933009569
+    AccessControlEnumerable(address(_domainPrice)).revokeRole(0x0, 0xAdc6a8fEB5C53303323A1D0280c0a0d5F2e1a14D);
+
+    // Remove operator role for RNS Unified
+    AccessControlEnumerable(address(_rns)).revokeRole(_rns.PAUSER_ROLE(), duke);
+    AccessControlEnumerable(address(_rns)).revokeRole(_rns.PROTECTED_SETTLER_ROLE(), duke);
+
+    // Remove operator role for RNS Auction
+    AccessControlEnumerable(address(_auction)).revokeRole(_auction.OPERATOR_ROLE(), duke);
+    // Remove operator role for RNS Domain Price
+    AccessControlEnumerable(address(_domainPrice)).revokeRole(_domainPrice.OVERRIDER_ROLE(), duke);
+    // Remove operator role for RNS Registrar Controller
+    AccessControlEnumerable(address(_ronController)).revokeRole(_ronController.OPERATOR_ROLE(), duke);
+    AccessControlEnumerable(address(_ronController)).revokeRole(_ronController.PAUSER_ROLE(), duke);
```

Comment:
Do we need to grant these roles to other address?

Reply:
granted to new wallet: https://app.roninchain.com/address/0x35558cffb77c6bdeed90cb582ef9404f2032f069